spring-integration-core/src/main/java/org/springframework/integration/json/JsonPropertyAccessor.java

```diff
@@ -16,38 +16,50 @@
 
 package org.springframework.integration.json;
 
-import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
 
 import org.springframework.expression.AccessException;
 import org.springframework.expression.EvaluationContext;
 import org.springframework.expression.PropertyAccessor;
 import org.springframework.expression.TypedValue;
-import org.springframework.util.Assert;
+import org.springframework.integration.support.CompositePropertyAccessor;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ArrayNode;
-import com.fasterxml.jackson.databind.node.ContainerNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 
 /**
- * A SpEL {@link PropertyAccessor} that knows how to read on Jackson JSON objects.
+ * A SpEL {@link CompositePropertyAccessor} that wraps node and String accessors.
+ *
+ * Also delegates to the appropriate accessor for backwards compatibility if used
+ * programmatically.
  *
  * @author Eric Bottard
  * @author Artem Bilan
+ * @author Gary Russell
  * @since 3.0
  */
-public class JsonPropertyAccessor implements PropertyAccessor {
+public class JsonPropertyAccessor implements PropertyAccessor, CompositePropertyAccessor {
```

Comment:
`PropertyAccessor` is redundant here, because `CompositePropertyAccessor extends PropertyAccessor`